src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +152,63 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
```

Comment:
This method assumes that the file is small. It would simplest (and also likely more performant) to:
- Open the file
- Gets it length
- Rent byte array of the right length from ArrayPool
- Read the file content
- Construct the string directly from the byte array

Reply:
> Gets it length

Hmm, not sure whether it is going to work for proc file system. 

Something along these lines...

Reply:
Yup, I have the fix coming up. I was running benchmarks after incorporating @stephentoub's feedback. Making the encoding and decoder local has improved the performance compared to reusabletextreader:

|     Method |        Job |                                                                       Toolchain |     Mean |    Error |   StdDev |   Median |      Min |      Max | Ratio | MannWhitney(3ms) |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------- |----------- |-------------------------------------------------------------------------------- |---------:|---------:|---------:|---------:|---------:|---------:|------:|----------------- |-------:|------:|------:|----------:|
| WorkingSet | Job-GSJGVE |   /runtime_base/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/corerun | 13.50 us | 0.080 us | 0.071 us | 13.49 us | 13.38 us | 13.66 us |  1.00 |             Same | 0.2680 |     - |     - |   2.26 KB |
| WorkingSet | Job-ELDEWS | /runtime_pr/artifacts/tests/coreclr/Linux.x64.Release/Tests/Core_Root/corerun | 13.39 us | 0.116 us | 0.103 us | 13.42 us | 13.20 us | 13.52 us |  0.99 |             Base | 0.2676 |     - |     - |   2.31 KB |

Reply:
I will try with length approach (to simplify the code and avoid re-renting).

Reply:
> not sure whether it is going to work for proc file system.

This is a good point. I just realized that during "reading the stream and counting the length, allocating, then re-reading from pos:0" the data might get overwritten by the OS and we may end up with wrong size. If that is the case, in order to avoid it, we would still need to maintain the loop-until structure. I am not sure if /proc/[pid]/* files can be read atomically, but with current implementation the risk seems to be lower.

Reply:
Hmm, that's interesting. What are the atomicity guarantees for the proc file system?  Is reading the file in multiple chunks reliable; or do you have to read the whole file in one chunk to make it fully reliable?

Reply:
I couldn't find much information on atomic reads, other than this old [SO thread](https://stackoverflow.com/questions/5713451/is-it-safe-to-parse-a-proc-file). Some answers sound scary, in the light of runtime's implementation of reads under `/proc`. One interesting piece of information is:

> All the kernel does is print its internal state into its own memory using a `sprintf`-like function, and that memory is copied into userspace whenever you issue a `read(2)` system call.
>
> The kernel handles these calls in an entirely different way than for regular files, which could mean that the entire snapshot of the data you will read could be ready at the time you `open(2)` it, while the kernel makes sure that concurrent calls are consistent and atomic. I haven't read that anywhere, but it doesn't really make sense to be otherwise.

if that is the case, then our chunk-based reads (here and in other places) appear to be (thread)safe.

Reply:
I have done some ad-hoc testing. It appears that `/proc` file system caches the response to provide consistent output and avoid tearing. We should be ok.